src/libhirte/common/config.h

```diff
@@ -0,0 +1,120 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#pragma once
+
+#include <stdbool.h>
+
+/*
+ * Existing configuration environment options
+ */
+#define ENV_LOG_LEVEL "HIRTE_LOG_LEVEL"
```

Comment:
Should these consts be exported? Are they not meant to be used only inside the config.c file? 
Furthermore, since they are basically a concatenation of `HIRTE_` and the name of the `CFG` flag, consider not defining them directly and instead use a concatenation macro (https://gcc.gnu.org/onlinedocs/cpp/Concatenation.html). This will make sure the names are aligned and will make it simpler to support the such flags

Reply:
I'm not a fan of using consts and macros from the code in the test code as it can hide issues. For example, a typo here will not show in the test.

Reply:
My bad, it was concatenation macros that did the trick, just usual macros.

If we agree that the tests should not use the macros from the code, then they are only used in one place. Therefore, instead of defining them, they can be created using macros and
in the `config.c` file's `cfg_load_from_env` you  will not need to maintain two arrays:
```
#define EXTERAL_PREFIX "HIRTE_"
#define ENV_FLAG_NAME(option_name) ( EXTERAL_PREFIX option_name)

int cfg_load_from_env(struct hashmap *config) {
        const char *option_names[] = { CFG_LOG_LEVEL, CFG_LOG_TARGET, CFG_LOG_IS_QUIET };

        int length = sizeof(option_names) / sizeof(option_names[0]);
        for (int i = 0; i < length; i++) {
                char *value = getenv(ENV_FLAG(option_names[i]);
                if (value != NULL && strlen(value) > 0) {
                        // environment variable is defined and has a non-empty value, so store it to the configuration
                        int result = cfg_set_value(config, option_names[i], value);
                        if (result != 0) {
                                return result;
                        }
                }
        }
        return 0;
}
``` 
  The nice thing here is that to add an support for a new external variable you just need to edit the `option_names` array

Reply:
With the differences between the current naming of configuration options and environment variables, it's no longer possible, but I've used the idea of having existing environment variable and relevant configuration options in arrays, so it's easy to add some new environment variable mapping